### PR TITLE
Fix fields indexing

### PR DESCRIPTION
**Pull request by dpgrote**
This makes two changes.

When specifying the global index in a fields wrapper, the code wasn't distinguishing between `0` and `None` when determining what data to fetch. This only matters with the ghost cells are included, since `0` means global index zero which does not include the ghost cells, and `None` would include all cells, including the ghost cells. This is fixed.

This also adds checks of the indexing, ensuring that they are within the bounds. This is important since the indexing is slightly different than numpy indexing, with negative indices behaving differently.

### CODE REVIEW

File: Python/pywarpx/fields.py [hunk 1]
```diff
@@ -140,6 +140,23 @@ def __getitem__(self, index):
         elif self.dim == 3:
             return self._getitem3d(index)
 
+    def _find_start_stop(self, ii, nn, ngrow, d):
+        if isinstance(ii, slice):
```
Comment (ax3l) [inline Python/pywarpx/fields.py:153]: Just a quick idea, can we add a parameter description for this helper function? :)
```suggestion
        """Short description

        ii : integer
            ...
        nn : integer
            ...
        ngrow : integer
            ...
        d : integer
            ...
        """
        if isinstance(ii, slice):
```

File: Python/pywarpx/fields.py [hunk 2]
```diff
@@ -140,6 +140,31 @@ def __getitem__(self, index):
         elif self.dim == 3:
             return self._getitem3d(index)
 
+    def _find_start_stop(self, ii, nn, ngrow, d):
+        """Given the input index, calculate the start and stop range of the indices.
+        - ii: index, either a slice object or an integer
+        - nn: the global number of cells along the specified direction
```
Comment (ax3l) [inline Python/pywarpx/fields.py:146]: Just one param skipped :)
```suggestion
        - nn: the global number of cells along the specified direction
        - ngrow: ...
```

Comment (dpgrote) [reply Python/pywarpx/fields.py:146]: That's what I get for hurrying :)

**Review by ax3l — APPROVED** [review]
Thank you, Dave! :rocket: :sparkles:

**Pull request merged**